handler-proxy/src/main/java/io/netty/handler/proxy/HttpProxyHandler.java

```diff
@@ -127,13 +144,20 @@ protected void removeDecoder(ChannelHandlerContext ctx) throws Exception {
     @Override
     protected Object newInitialMessage(ChannelHandlerContext ctx) throws Exception {
         InetSocketAddress raddr = destinationAddress();
-        final String host = NetUtil.toSocketAddressString(raddr);
+
+        String hostString = HttpUtil.formatForHttp(raddr);
+        int port = raddr.getPort();
+        String url = hostString + ":" + port;
+        String hostHeader = (ignoreDefaultPortsInConnectHostHeader && (port == 80 || port == 443)) ?
```

Comment:
@slandelle can you do this ?

Reply:
I'm not convinced it would make sense to test this single line of code.
Then, url and Host header computations are entangled so I can save allocations.
I can't test `HttpProxyHandler#newInitialMessage` because I would need to override `destinationAddress()` which is final.
I'm going to extract CONNECT request creation in a static method.

Reply:
Actually, it should be fine. Stay tuned.